src/tools/non-promotion-reasons.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Fixing type promotion failures
+description: Solutions for cases where you know more about a field's type than Dart can determine.
+toc: false
+---
+
+{{ site.alert.warn}}
+  **This page is under construction**
+  ([issue #2940][]).
+{{ site.alert.end}}
+
+This page will have information to help you understand
+why type promotion failures occur,
```

Comment:
I'm a bit confused about the word 'failure' here. To me that sounds a bit like something that might occur when there is a bug in the analyzer... Is there a another word that might make it more clear that this isn't happening simply because it's something we're currently not capable of doing?